### PR TITLE
Feature/extensible trainer

### PR DESCRIPTION
**Pull request by sei-nwinski**

### CODE REVIEW

File: juneberry/documentation/dataset_configuration_specification.md [hunk 1]
```diff
@@ -162,7 +177,6 @@ the subsequent row is passed to the transformer.
 
 ### algorithm
 This can be:
-* none - All images are used.
```
Comment (sei-amellinger) [inline juneberry/documentation/dataset_configuration_specification.md:165]: This should not be removed

File: juneberry/pytorch/evaluator.py [hunk 2]
```diff
@@ -55,37 +33,39 @@
...
+class Evaluator(EvaluatorBase):
     """
-    This subclass is the Pytorch-specific version of the Evaluator.
+    This subclass is the Pytorch-specific version of the EvaluatorBase.
     """
 
-    def __init__(self, lab: Lab, model_config: ModelConfig, dataset: DatasetConfig, model_manager: ModelManager,
-                 eval_dir_mgr: EvalDirMgr, eval_options: SimpleNamespace = None):
+    def __init__(self, lab: Lab, model_config: ModelConfig, model_manager: ModelManager, eval_dir_mgr: EvalDirMgr,
+                 dataset: DatasetConfig, eval_options: SimpleNamespace = None, log_file: str = None, **kwargs):
```
Comment (sei-amellinger) [inline juneberry/pytorch/evaluator.py:53]: I think kwargs is unused.  Should we be using it?

File: scripts/run_system_test.py [hunk 3]
```diff
@@ -423,8 +423,7 @@ def get_model_train_file_patterns(model_name: str) -> list:
     if model_name in ["imagenette_224x224_rgb_unit_test_tf_resnet50"]:
         return files
 
-    if model_name in ["imagenette_160x160_rgb_unit_test_pyt_resnet18",
-                      "tabular_binary_sample"]:
+    if model_name in ["imagenette_160x160_rgb_unit_test_pyt_resnet18", "tabular_binary_sample"]:
```
Comment (sei-amellinger) [inline scripts/run_system_test.py:426]: Are these sorts of changes necessary?

**Comment by sei-nwinski** [comment]
PR 23 replaces this one.

**Pull request closed without merging**